app/src/main/java/ch/epfl/qedit/backend/auth/AuthenticationService.java

```diff
@@ -1,16 +1,19 @@
 package ch.epfl.qedit.backend.auth;
 
+import ch.epfl.qedit.R;
 import ch.epfl.qedit.model.User;
 import ch.epfl.qedit.util.Callback;
+import ch.epfl.qedit.util.Error;
 import ch.epfl.qedit.util.Response;
 
 /**
  * Represents an authentication service, that is a service to which login requests can be sent. It
  * handles asynchronous responses by using a callback.
  */
 public interface AuthenticationService {
-    int CONNECTION_ERROR = 1;
-    int WRONG_TOKEN = 2;
+
+    Error CONNECTION_ERROR = new Error(R.string.connection_error_message);
```

Comment:
What is `Error`, why not use an `int` ?

Reply:
It's brand new! It is use to refactor the code when the error is a fail (show a toast etc...) it make the use of error responce simpler